### PR TITLE
Refactoring client

### PR DESCRIPTION
**Pull request by isSerge**
- `Client` module refactoring
- move polkadot/apps link from client to `farmedList` component (since it belongs to presentation logic)
- fetch initial peers count value when Dashboard mounts and increase timeout for refetching
- other minor improvements

### CODE REVIEW

**Review by ozgunozerk — APPROVED** [review]
I like the changes overall, much better names then previous code 👍

File: src/components/farmedList.vue [hunk 1]
```diff
@@ -51,9 +51,9 @@ q-card(bordered flat)
         q-separator
         .row.q-pa-xs.q-ml-sm.q-ma-xs
           .col-4
-            p {{ block.rewardAddr.substring(0, 8) + '...' + block.rewardAddr.substring(block.rewardAddr.length - 8, block.rewardAddr.length) }}
+            p {{ formatAddress(block.rewardAddr) }}
           .col-3
-            a(:href="block.appsLink" target="_blank") # {{ block.blockNum.toLocaleString() }}
+            a(:href="blockLink(block.blockNum.toString())" target="_blank") # {{ block.blockNum.toLocaleString() }}
```
Comment (nazar-pc) [inline src/components/farmedList.vue:56]: Would read cleaner if it took block number as a number in arguments IMO

File: src/lib/client.ts [hunk 2]
```diff
@@ -32,110 +28,89 @@ export class Client {
...
+  async getBlockRewards(hash: IU8a): Promise<number> {
+    let blockReward = 0;
+    const apiAt = await this.api.at(hash);
+    const records = (await apiAt.query.system.events()) as Vec<EventRecord>;
+    records.forEach((record) => {
+      const { section, method, data } = record.event
+      if (section === "rewards" && method === "BlockReward") {
+        const reward = this.api.registry.createType("u128", data[1]);
+        blockReward = Number((reward.toBigInt() * 100n) / SUNIT) / 100;
+      } else if (section === "transactionFees") {
```
Comment (nazar-pc) [inline src/lib/client.ts:52]: Yes, rewards pallet has a few different rewards all in one place, including storage and compute fees (but those are negligible).

File: src/lib/client.ts [hunk 3]
```diff
@@ -32,110 +28,89 @@ export class Client {
...
+    return blockReward;
+  }
+
+  async startSubscription(): Promise<void> {
+    const rewardAddress: string = (await appConfig.read()).rewardAddress;
+
+    this.unsubscribe = await this.api.rpc.chain.subscribeNewHeads(
+      async ({ hash, number }) => {
+        const blockNum = number.toNumber()
+        const signedBlock = await this.api.rpc.chain.getBlock(hash)
```
Comment (nazar-pc) [inline src/lib/client.ts:66]: No need to download the whole block if you need just a header

File: src/lib/client.ts (hunk 2 above)
Comment (isSerge) [reply src/lib/client.ts:52]: I was confused about that mysterious TODO, thanks for clarifying

File: src/lib/client.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ export class Client {
         const reward = this.api.registry.createType("u128", data[1]);
         blockReward = Number((reward.toBigInt() * 100n) / SUNIT) / 100;
       } else if (section === "transactionFees") {
-        // TODO
+        // TODO: include storage and compute fees
```
Comment (nazar-pc) [inline src/lib/client.ts:53]: And voting :slightly_smiling_face:

File: src/lib/client.ts [hunk 5]
```diff
@@ -63,16 +63,19 @@ export class Client {
     this.unsubscribe = await this.api.rpc.chain.subscribeNewHeads(
       async ({ hash, number }) => {
         const blockNum = number.toNumber()
-        const signedBlock = await this.api.rpc.chain.getBlock(hash)
-        const preRuntimeLog = signedBlock.block.header.digest.logs.find((digestItem) => digestItem.isPreRuntime)?.asPreRuntime[1];
+        const header = await this.api.rpc.chain.getHeader(hash);
+
+        // TODO: handle vote rewards: check VoteReward events and aggregate
```
Comment (nazar-pc) [inline src/lib/client.ts:68]: Voting rewards are in the transactions and you shouldn't need to read blocks for that, if you can just go through events it'll tell you everything you need to know.

**Review by nazar-pc — APPROVED** [review]

File: src/lib/client.ts (hunk 4 above)
Comment (isSerge) [reply src/lib/client.ts:53]: Voting is already in progress in another branch

**Pull request merged**